src/result.rs

```diff
@@ -1,9 +1,74 @@
 use std::convert::From;
-use std::fmt::{Debug, Error};
+use std::fmt::{Debug, Display, Error};
 use std::{fmt, io};
 
 use thiserror::Error;
 
+/// Represents the position, as line and column, within a text format ion document.
+#[derive(Clone, Debug)]
+pub enum TextPosition {
+    /// No text position available; implies binary format is being read.
+    None,
+    /// The position represented as line and column.
+    LineColumn(usize, usize),
+}
+
+#[derive(Clone, Debug)]
+pub struct Position {
+    pub byte_offset: usize,
+    pub line_column: TextPosition,
+}
+
+impl Position {
+    fn with_offset(offset: usize) -> Self {
+        Position {
+            byte_offset: offset,
+            line_column: TextPosition::None,
+        }
+    }
+
+    fn with_line_info(offset: usize, line: usize, column: usize) -> Self {
```

Comment:
This is the only place the phrase `line_info` appears, which makes me wonder if this should be the slightly wordier `with_line_and_column` or perhaps `with_text_position`?

Reply:
Added `with_text_position` as a builder-style method in the latest commit.